src/d3d9/d3d9_initializer.cpp

```diff
@@ -131,18 +131,41 @@ namespace dxvk {
     // If the buffer is mapped, we can write data directly
     // to the mapped memory region instead of doing it on
     // the GPU. Same goes for zero-initialization.
-    for (uint32_t i = 0; i < pTexture->CountSubresources(); i++) {
-      DxvkBufferSliceHandle mapSlice  = pTexture->GetBuffer(i)->getSliceHandle();
-
-      if (pInitialData != nullptr) {
-        std::memcpy(
-          mapSlice.mapPtr,
-          pInitialData,
-          mapSlice.length);
-      } else {
-        std::memset(
-          mapSlice.mapPtr, 0,
-          mapSlice.length);
+    const D3D9_COMMON_TEXTURE_DESC* desc = pTexture->Desc();
+    for (uint32_t a = 0; a < desc->ArraySize; a++) {
+      for (uint32_t m = 0; m < desc->MipLevels; m++) {
+        uint32_t subresource = pTexture->CalcSubresource(a, m);
+        DxvkBufferSliceHandle mapSlice  = pTexture->GetBuffer(subresource)->getSliceHandle();
+
+        if (pInitialData != nullptr) {
+          VkExtent3D mipExtent = pTexture->GetExtentMip(m);
+          const DxvkFormatInfo* formatInfo = imageFormatInfo(pTexture->GetFormatMapping().FormatColor);
+          VkExtent3D blockCount = util::computeBlockCount(mipExtent, formatInfo->blockSize);
+          uint32_t pitch = blockCount.width * formatInfo->elementSize;
+          uint32_t alignedPitch = align(pitch, 4);
+
+          D3DRESOURCETYPE resourceType = pTexture->GetType();
+          VkImageType type = resourceType == D3DRTYPE_VOLUME || resourceType == D3DRTYPE_VOLUMETEXTURE
```

Comment:
pTexture->GetImage()->info().type

Reply:
That doesn't work because this is only called for POOL_SYSTEMMEM textures which don't have an image.